### PR TITLE
ROX-11227: Add X-Frame-Options header to UI request

### PR DESCRIPTION
**Pull request by theencee**
## Description

Don't allow anyone to iframe it

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by SimonBaeumer — DISMISSED** [review]

**Review by vjwilson — CHANGES_REQUESTED** [review]
In double-checking my references, I found that the second header I recommended now has better support than the first.

File: central/ui/mux.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ func Mux() http.Handler {
 		http.ServeFile(w, r, "/ui/service-worker.js")
 	})
 	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("X-Frame-Options", "sameorigin")
```
[the remark this answers is not in the document]
Comment (theencee) [reply central/ui/mux.go:19]: I guess my understanding of CSP headers was pretty outdated. Didn't realize it has more support than XFO. Added.

**Review by vjwilson — APPROVED** [review]
Tested on a live server, including the Test Auth Provider workflow.

**Pull request merged**